builtin/credential/ldap/path_config.go

```diff
@@ -91,20 +112,38 @@ func pathConfig(b *backend) *framework.Path {
 	}
 }
 
+/*
+ * Construct ConfigEntry struct using stored configuration.
+ */
 func (b *backend) Config(req *logical.Request) (*ConfigEntry, error) {
-	entry, err := req.Storage.Get("config")
+	// Schema for ConfigEntry
+	fd, err := b.getConfigFieldData()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new ConfigEntry, filling in defaults where appropriate
+	result, err := b.newConfigEntry(fd)
+	if err != nil {
+		return nil, err
+	}
+
+	storedConfig, err := req.Storage.Get("config")
 	if err != nil {
 		return nil, err
 	}
-	if entry == nil {
+
+	if storedConfig == nil {
```

Comment:
Shouldn't this return `result` rather than `nil`, since then it'll return a default entry?